### PR TITLE
[ Fabric ] Removing organization from existing Fabric network

### PR DESCRIPTION
**Pull request by rahultoraskarr**
Signed-off-by: rahultoraskarr <rahul.p.toraskar@accenture.com>

**Changelog**
- Add functionality to remove an organization from existing fabric network using the playbook remove-organization.yaml

 

**Reviewed by**
@jagpreetsinghsasan @lakshyakumar

 

**Linked issue**
#1101

### CODE REVIEW

File: docs/source/operations/removing_org_fabric.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<a name = "adding-new-org-to-existing-network-in-fabric"></a>
```
Comment (lakshyakumar) [inline docs/source/operations/removing_org_fabric.md:1]: anchor says "adding-new-org-to-existing-network-in-fabric"

File: docs/source/operations/removing_org_fabric.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          org_status: delete  # delete for to be deleted organization(s)
+        - organization:
+          ..
+          ..
+          org_status: existing  # existing for old organization(s)
+
+The `network.yaml` file should contain the specific `network.organization` patch along with the orderer information.
+
+
+For reference, see `network-fabric-add-organization.yaml` file [here](https://github.com/hyperledger-labs/blockchain-automation-framework/tree/master/platforms/hyperledger-fabric/configuration/samples).
```
Comment (lakshyakumar) [inline docs/source/operations/removing_org_fabric.md:56]: refering to "network-fabric-add-organization.yaml"

File: docs/source/operations/removing_org_fabric.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+---
+
+<a name = "create_config_file"></a>
+## Modifying Configuration File
+
+Refer [this guide](./fabric_networkyaml.md) for details on editing the configuration file.
+
+While modifying the configuration file(`network.yaml`) for removing an organization, all the existing organizations should have `org_status` tag as `existing` and the new organization should have `org_status` tag as `delete` under `network.channels` eg.
+
```
Comment (lakshyakumar) [inline docs/source/operations/removing_org_fabric.md:24]: it says "the new organization should have"

File: docs/source/operations/removing_org_fabric.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+# Removing an organization in Hyperledger Fabric
+
+- [Prerequisites](#prerequisites)
+- [Modifying configuration file](#create_config_file)
+- [Running playbook to deploy Hyperledger Fabric network](#run_network)
+
+
+<a name = "prerequisites"></a>
+## Prerequisites
+To add a new organization a fully configured Fabric network must be present already, i.e. a Fabric network which has Orderers, Peers, Channels (with all Peers already in the channels). The corresponding crypto materials should also be present in their respective Hashicorp Vault.
```
Comment (lakshyakumar) [inline docs/source/operations/removing_org_fabric.md:11]: prerequisite is for adding a new org

File: docs/source/operations/removing_org_fabric.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+The `network.yaml` file should contain the specific `network.organization` patch along with the orderer information.
+
+
+For reference, see `network-fabric-add-organization.yaml` file [here](https://github.com/hyperledger-labs/blockchain-automation-framework/tree/master/platforms/hyperledger-fabric/configuration/samples).
+
+<a name = "run_network"></a>
+## Run playbook
+
+The [add-new-organization.yaml](https://github.com/hyperledger-labs/blockchain-automation-framework/platforms/hyperledger-fabric/configuration/remove-organization.yaml) playbook is used to remove organization(s) from the existing network. This can be done using the following command
```
Comment (lakshyakumar) [inline docs/source/operations/removing_org_fabric.md:61]: the playbook is for addition of new organization

File: platforms/hyperledger-fabric/configuration/roles/create/delete_org_script/Readme.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+## ROLE: create/new_organization/create_block
```
Comment (lakshyakumar) [inline platforms/hyperledger-fabric/configuration/roles/create/delete_org_script/Readme.md:1]: change "new_organization"

**Review by lakshyakumar — CHANGES_REQUESTED** [review]

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM. I have already reviewed it (when this was demoed) and the changes are already in place.

**Review by lakshyakumar — APPROVED** [review]

**Pull request merged**